dynamic_instrumentation/src/instrument.rs

```diff
@@ -229,25 +230,21 @@ impl<'tcx> Visitor<'tcx> for InstrumentationAdder<'_, 'tcx> {
                 }
             }
             _ if !is_region_or_unsafe_ptr(value_ty) => {}
-            Rvalue::AddressOf(_, p)
-                if has_outer_deref(p)
-                    && place_ty(&remove_outer_deref(*p, self.tcx())).is_region_ptr() =>
-            {
-                let source = remove_outer_deref(*p, self.tcx());
-                // Instrument which local's address is taken
-                self.loc(location.successor_within_block(), copy_fn)
-                    .arg_var(dest)
-                    .source(&source)
-                    .dest(&dest)
-                    .debug_mir(location)
-                    .add_to(self);
-            }
             Rvalue::AddressOf(_, p) => {
                 // Instrument which local's address is taken
-                self.loc(location.successor_within_block(), addr_local_fn)
+                let source = try_remove_outer_deref(*p, self.tcx())
```

Comment:
i don't want to combine these, there are more complicated cases incoming and it'll be much clearer if they're separate